### PR TITLE
Switch counter metrics to use CUMULATIVE

### LINKED ISSUE #18: Switch counter metrics to use cumulative or delta
It turns out that a counter metrics use a gauge metric type as [here](https://github.com/google/go-metrics-stackdriver/blob/main/stackdriver.go#L368)

```
MetricKind: metricpb.MetricDescriptor_GAUGE,
```

This doesn't seem to be right. According to documentation of [Aligner](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.alertPolicies#Aligner) I am unable to calculate deltas if the metric is not defined as delta or cumulative. This is a  big limitation. Should we consider switching to cumulative in case of counters?

### PR DESCRIPTION
**Pull request by prog8**
fixes #18

### CODE REVIEW

File: stackdriver.go [hunk 1]
```diff
@@ -365,7 +365,7 @@ func (s *Sink) report(ctx context.Context) {
 				Type:   fmt.Sprintf("custom.googleapis.com/%s%s", s.prefix, name),
 				Labels: labels,
 			},
-			MetricKind: metricpb.MetricDescriptor_GAUGE,
+			MetricKind: metricpb.MetricDescriptor_CUMULATIVE,
 			Resource:   resource,
 			Points: []*monitoringpb.Point{
 				{
```
Comment (tam7t) [inline stackdriver.go:371]: I think for CUMULATIVE the points also need to have a StartTime:

https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors#metrickind

**Comment by prog8** [comment]
@tam7t let me know what are your findings when you do simple backwards compatibility testing

**Comment by rf** [comment]
So I tested this and I'm getting this error
`Failed to write time series data: rpc error: code = InvalidArgument desc = Field timeSeries[0].points[0].interval.start_time had an invalid value of "2021-08-31T14:19:47-07:00": The start time must be before the end time (2021-08-31T14:19:47-07:00) for the non-gauge metric 'custom.googleapis.com/go-metrics/my-metric-id`
and yes, this is a backwards incompatible change -- I had to manually delete the old metric since the type changed.

**Comment by rf** [comment]
The correct `MetricKind` may actually be `DELTA`.

**Comment by prog8** [comment]
@rf thanks for your effort. 
@tam7t would you accept PR afte changing `MetricKind` to `DELTA`?

**Comment by tam7t** [comment]
https://github.com/google/go-metrics-stackdriver/pull/19#discussion_r686977500 is the reason for the error there. I'm still evaluating the effects of changing the metric types at all but i do not suspect that `DELTA` will make it more difficult as additional time periods would need to be accounted for

**Comment by rf** [comment]
It turns out `DELTA` metrics are not supported in Stackdriver for custom metrics.

I'm not sure that counters in `go-metrics` are intended to be treated like a statsd counter (i.e. something that resets after an interval). It's not clear if other sinks treat it this way.

**Comment by tam7t** [comment]
@rf I believe that what you are describing is how to implement and report a `DELTA` metric (clearing each interval _and_ reporting the start/end time of the interval) because you are reporting the _change_ of the counter over a specific time interval. To report a counter as `CUMULATIVE` you need to report the start time of the accumulation in the report to stackdriver, as the histogram metrics do.

**Comment by rf** [comment]
I did have some trouble getting `CUMULATIVE` to work -- I was also trying to use a `StartTime` that was basically the end of the last reporting interval. Not sure if that was right, should I be using the same start time as for histograms?

I was able to get this to work, however, by using `GAUGE` and just resetting the counters every time we do the deep copy in order to flush. It has the benefit of being somewhat backwards incompatible (because the metrics don't need to be re-created) but will end up reporting very different values.

**Pull request closed without merging**